File: packages/server-core/src/lib/movies/entities/movie-entity.ts
import { EntityBase } from '../../common/entities/entity-base';
import { Column, Entity, OneToMany } from 'typeorm';
import { MovieRoleEntity } from './movie-role-entity';
import { MovieRole } from './movie-role';
import { PersonEntity } from './person-entity';

@Entity()
export class MovieEntity extends EntityBase {
  @Column('string')
  title: string;

  @Column('string')
  description: string;

  @Column('date')
  releaseDate: Date;

  @OneToMany(() => MovieRoleEntity, (role) => role.movie)
  roles: MovieRoleEntity[] = [];

  @Column('string')
  filePath: string;

  addPerson(role: MovieRole, person: PersonEntity) {
    const movieRole = MovieRoleEntity.create(role, person, this);
    this.roles = [...this.roles, movieRole];
  }

  addRole(role: MovieRoleEntity) {
    this.roles = [...this.roles, role];
  }
}

File: src/House.Flix.Core/Movies/Entities/MovieEntity.cs
using House.Flix.Core.Common.Entities;
using House.Flix.Core.People.Entities;

namespace House.Flix.Core.Movies.Entities;

public class MovieEntity : IEntity
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Plot { get; set; }
    public string Rating { get; set; }
    public string VideoFilePath { get; set; }

    public List<MovieRoleEntity> MovieRoles { get; set; } = new();

    public void AddRole(PersonEntity person, RoleEntity role)
    {
        MovieRoles.Add(
            new MovieRoleEntity
            {
                Movie = this,
                Person = person,
                Role = role
            }
        );
    }
}
